src/lib/core/CHIPTLVWriter.cpp

```diff
@@ -629,24 +629,21 @@ CHIP_ERROR TLVWriter::WriteElementHead(TLVElementType elemType, Tag tag, uint64_
     else
         p = stagingBuf;
 
-    if (IsSpecialTag(tag))
+    if (IsContextTag(tag))
```

Comment:
Why were the changes here needed?

Reply:
We have a function for checking if the tag is a context tag, so I thought it's better to use it like in other places instead of rely on the knowledge of internal encoding (that context tag is a special tag with low tag numbers). In fact, probably `IsSpecialTag` below should also be replaced with `IsAnonymousTag` for clarity. I can revert the changes though if that's controversial.

Reply:
OK, that makes sense.  We should consider making IsSpecialTag return false for context tags, actually....